src/WebJobs.Script/Host/FunctionAppValidationService.cs

```
@@ -0,0 +1,79 @@
+                Utility.ExecuteAfterColdStartDelay(_environment, Validate, cancellationToken);
+            }
+
+            await Task.CompletedTask;
+        }
+
+        private void Validate()
+        {
+            try
+            {
+                if (_scriptOptions.Value.RootScriptPath is not null &&
+                    !_scriptOptions.Value.IsDefaultHostConfig &&
+                    Utility.IsDotnetIsolatedApp(environment: _environment) &&
+                    !Directory.Exists(Path.Combine(_scriptOptions.Value.RootScriptPath, ScriptConstants.AzureFunctionsSystemDirectoryName)))
+                {
+                    string azureFunctionsDirPath = Path.Combine(_scriptOptions.Value.RootScriptPath, ScriptConstants.AzureFunctionsSystemDirectoryName);
+
+                    // Search for the .azurefunctions directory within nested directories to verify scenarios where it isn't located at the root. This situation occurs when a function app has been improperly zipped.
+                    IEnumerable<string> azureFunctionsDirectories = Directory.GetDirectories(_scriptOptions.Value.RootScriptPath, ScriptConstants.AzureFunctionsSystemDirectoryName, SearchOption.AllDirectories)
+                        .Where(dir => !dir.Equals(azureFunctionsDirPath, StringComparison.OrdinalIgnoreCase));
+
+                    if (azureFunctionsDirectories.Any())
+                    {
+                        string azureFunctionsDirectoriesPath = string.Join(", ", azureFunctionsDirectories).Replace(_scriptOptions.Value.RootScriptPath, string.Empty);
+                        _logger.IncorrectAzureFunctionsFolderPath(azureFunctionsDirectoriesPath);
+                    }
+                    else
+                    {
+                        _logger.MissingAzureFunctionsFolder();
+                    }
+                }
+            }
+            catch
+            {
+                // Ignore any exceptions.
```

Is there a specific reason for silently ignoring exceptions here? If so, we should add a comment about that here. Would it be better to log the errors, so we’re aware of unexpected behavior (if any) ?.